src/uct/ib/base/ib_md.c

```diff
@@ -1411,6 +1451,58 @@ uct_ib_md_parse_subnet_prefix(const char *subnet_prefix_str,
     return UCS_OK;
 }
 
+static double uct_ib_md_read_pci_bw(struct ibv_device *ib_device)
+{
+    const char pci_data_path_fmt[]   = "/sys/class/infiniband/%s/device/%s";
+    const char pci_width_file_name[] = "current_link_width";
+    const char pci_speed_file_name[] = "current_link_speed";
+    char pci_width_str[16];
+    char pci_speed_str[16];
+    uct_ib_md_pci_info_t *p;
+    double bw;
+    unsigned width;
+    ssize_t len;
+    size_t i;
+
+    len = ucs_read_file(pci_width_str, sizeof(pci_width_str) - 1, 1, pci_data_path_fmt,
+                        ib_device->name, pci_width_file_name);
+    if (len < 1) {
+        return DBL_MAX; /* failed to read file */
+    } else {
```

Comment:
minor: you don't really need `else`, because you return in error case

Reply:
removed 'else'